### PR TITLE
Added the NumberListAttribute attribute

### PR DESCRIPTION
**Pull request by digitizdat**
This adds a List attribute that only supports numbers (i.e. integers). In general, DynamoDB will support a heterogeneous list of types under the List ('L') type:

AWS Documentation » Amazon DynamoDB » API Reference » Data Types » [AttributeValue](http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_AttributeValue.html)

But, developing that more general case will take additional time to build and test. For now, this very 
specific attribute will answer the needs of some PynamoDB users.

### CODE REVIEW

**Comment by digitizdat** [comment]
@jlafon, would you please merge this PR to add the NumberListAttribute attribute, and to address issue #122?

**Comment by jlafon** [comment]
@digitizdat apologies for the slow response. I'll review this now.

File: pynamodb/attributes.py [hunk 1]
```diff
@@ -239,6 +239,48 @@ def deserialize(self, value):
...
+
+    @staticmethod
+    def serialize(values):
+        """
+        Encode the given list of numbers into a list of AttributeValue types.
+        """
+        rval = []
+        for v in values:
+            if type(v) is list:
+                rval += [{'L': NumberListAttribute.serialize(v)}]
```
Comment (jlafon) [inline pynamodb/attributes.py]: Would you mind using the constants for 'L' and 'N' here to keep the style consistent?

File: pynamodb/tests/test_attributes.py [hunk 2]
```diff
@@ -402,6 +402,62 @@ def test_boolean_deserialize(self):
         self.assertEqual(attr.deserialize('0'), False)
 
 
+class NumberListAttributeTestCase(TestCase):
```
Comment (jlafon) [inline pynamodb/tests/test_attributes.py]: Thanks for including tests!

File: pynamodb/attributes.py (hunk 1 above)
Comment (digitizdat) [inline pynamodb/attributes.py]: Oh, of course.

**Comment by digitizdat** [comment]
@jlafon Defined constants are now used in place of the actual characters. Let me know if there's anything else.

**Comment by digitizdat** [comment]
@jlafon Was this ^^ the change you were looking for in your request that I use constants for 'L' and 'N' to keep the style consistent?

**Comment by digitizdat** [comment]
@jlafon Just checking in to see if this can be merged yet. I have added all of the most recent changes up to today, which did include some conflicts I had to manually resolve.

I would still like to get this PR merged so that I don't need to keep manually updating my fork.

**Comment by digitizdat** [comment]
Closing this PR to open it with a new PR that includes a cleaner implementation of NumberListAttribute that also includes an implementation of UnicodeListAttribute.

**Pull request closed without merging**